Content.Server/Storage/EntitySystems/StorageSystem.cs

```diff
@@ -316,7 +316,8 @@ private async void AfterInteract(EntityUid uid, ServerStorageComponent storageCo
                 {
                     if (entity == args.User
                         || !itemQuery.HasComponent(entity)
-                        || !_interactionSystem.InRangeUnobstructed(args.User, entity))
+                        || !_interactionSystem.InRangeUnobstructed(args.User, entity)
+                        || !CanInsert(uid, entity, out _, storageComp))
```

Comment:
You should do the expensive checks last.

Reply:
Ah, didn't think that InRangeUnobstructed would be more expensive, but yeah, all that raycasting couldn't possibly be faster than checking the existence of a couple of components. Thank you!